pkg/commands/images.go

```diff
@@ -0,0 +1,106 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021 Red Hat, Inc.
+ */
+
+package commands
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"github.com/k8stopologyawareschedwg/deployer/pkg/images"
+)
+
+type imagesOptions struct {
+	jsonOutput bool
+	rawOutput  bool
+}
+
+func NewImagesCommand(commonOpts *CommonOptions) *cobra.Command {
+	opts := &imagesOptions{}
+	images := &cobra.Command{
+		Use:   "images",
+		Short: "dump the container images used to deploy",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			imo := newImageOutput()
+			if opts.rawOutput {
```

Comment:
We actually have 4 possible permutations here.
But from the command line perspective, it looks like the flags are mutually exclusive.

Reply:
Thanks, I'll clarify a bit the help text. The flag will indeed produce 2*2=4 different outputs:
```
$ ./_out/deployer images
TAS_SCHEDULER_PLUGIN_IMAGE=k8s.gcr.io/scheduler-plugins/kube-scheduler:v0.19.9
TAS_SCHEDULER_PLUGIN_CONTROLLER_IMAGE=k8s.gcr.io/scheduler-plugins/controller:v0.19.9
TAS_RESOURCE_EXPORTER_IMAGE=quay.io/openshift-kni/resource-topology-exporter:4.9-snapshot

$ ./_out/deployer images -r
quay.io/openshift-kni/resource-topology-exporter:4.9-snapshot
k8s.gcr.io/scheduler-plugins/kube-scheduler:v0.19.9
k8s.gcr.io/scheduler-plugins/controller:v0.19.9

$ ./_out/deployer images --json
{"topology_updater":"quay.io/openshift-kni/resource-topology-exporter:4.9-snapshot","scheduler_plugin":"k8s.gcr.io/scheduler-plugins/kube-scheduler:v0.19.9","scheduler_controller":"k8s.gcr.io/scheduler-plugins/controller:v0.19.9"}

$ ./_out/deployer images --json -r
["quay.io/openshift-kni/resource-topology-exporter:4.9-snapshot","k8s.gcr.io/scheduler-plugins/kube-scheduler:v0.19.9","k8s.gcr.io/scheduler-plugins/controller:v0.19.9"]
```